packages/dialect-react-ui/components/Chat/index.tsx

```diff
@@ -75,87 +51,119 @@ function InnerChat({
           showThreadSettings(navigate, threadId),
       },
     });
-  }, [configure, isSolanaConnected, isWalletConnected, navigate]);
-
-  const someBackendConnected =
-    (isSolanaShouldConnect && isSolanaConnected) ||
-    (isDialectCloudShouldConnect && isDialectCloudConnected);
+  }, [configure, navigate]);
 
   useEffect(
     function pickRoute() {
-      if (!someBackendConnected) {
-        navigate(RouteName.NoConnection);
-      } else if (!isWalletConnected) {
-        navigate(RouteName.NoWallet);
-      } else if (isSigning) {
+      if (isSigning) {
         navigate(RouteName.SigningRequest);
       } else if (isEncrypting) {
         navigate(RouteName.EncryptionRequest);
       } else {
         navigate(RouteName.Main, { sub: { name: MainRouteName.Thread } });
       }
     },
-    [
-      navigate,
-      someBackendConnected,
-      isSolanaConnected,
-      isWalletConnected,
-      isSigning,
-      isEncrypting,
-    ]
+    [navigate, isSigning, isEncrypting]
   );
 
+  return (
+    <>
+      <Route name={RouteName.SigningRequest}>
+        <SignMessageInfo />
+      </Route>
+      <Route name={RouteName.EncryptionRequest}>
+        <EncryptionInfo />
+      </Route>
+      <Route name={RouteName.Main}>
+        <Main />
+      </Route>
+    </>
+  );
+}
+
+interface ChatProps {
+  dialectId: string;
+  type: ChatType;
+  wrapperClassName?: string;
+  contentWrapperClassName?: string;
+  onChatClose?: () => void;
+  onChatOpen?: () => void;
+}
+
+export default function Chat({
+  wrapperClassName,
+  contentWrapperClassName,
+  onChatOpen,
+  onChatClose,
+  ...props
+}: ChatProps) {
+  const { dialectId, type } = props;
+
   const { colors, modal, slider } = useTheme();
 
+  const {
+    connected: {
+      solana: {
+        connected: isSolanaConnected,
+        shouldConnect: isSolanaShouldConnect,
+      },
+      dialectCloud: {
+        connected: isDialectCloudConnected,
+        shouldConnect: isDialectCloudShouldConnect,
+      },
+    },
+  } = useDialectConnectionInfo();
+
+  const { connected: isWalletConnected } = useDialectWallet();
+
+  const someBackendConnected =
+    (isSolanaShouldConnect && isSolanaConnected) ||
+    (isDialectCloudShouldConnect && isDialectCloudConnected);
+
+  const hasError = !isWalletConnected || !someBackendConnected;
```

Comment:
Just realizing this now, but would be more convenient to have this aggregated information in DialectProvider, so that when developing a UI, a developer won't have to think about what exactly is wrong, and/or which backend is causing this issue. Wdyt?

No need to adjust this now, just a open question.

Reply:
Hm, I'm not 100% about it. Cause we already have all necessary information, and this is very app specific i thinnk..